### PR TITLE
Kds 754/time picker filter

### PR DESCRIPTION
**Pull request by nadeemaajihil**
Filtering

### CODE REVIEW

File: packages/time-picker/src/TimePicker/utils/time.ts [hunk 1]
```diff
@@ -25,19 +25,42 @@ export const formatDateToTime = (date: Date, locale, timeZone): string =>
...
+  if (isNaN(parseInt(aNum, 10))) {
+    return false
+  }
+  return true
+}
+
+const getDateRegEx = (segments: DateSegment[] | undefined) => {
+  let regex = ""
+  if (segments) {
+    for (const [, { type, text, isPlaceholder }] of Object.entries(segments)) {
```
Comment (jshli) [inline packages/time-picker/src/TimePicker/utils/time.ts:38]: Why apply `Object.entries` on an array? 
Is there a way we write this without using a `for...of` loop as loops aren't a common practice within the codebase? 

Not super tested, but either:
```suggestion
     const regex = segments
    .filter(
      segment =>
        ((segment.type === "literal" || segment.type === "dayPeriod") &&
          segment.isPlaceholder !== undefined &&
          !segment.isPlaceholder) ||
        isNumber(segment.text)
    )
    .map(segment => segment.text)
    .join("")
```

or if you don't like multiple loops:
```suggestion
   const regex = segments.reduce((regexString, segment) => {
    if (
      ((segment.type === "literal" || segment.type === "dayPeriod") &&
        segment.isPlaceholder !== undefined &&
        !segment.isPlaceholder) ||
      isNumber(segment.text)
    ) {
      regexString += segment.text
    }
    return regexString
  }, "")

```

Comment (nadeemaajihil) [reply packages/time-picker/src/TimePicker/utils/time.ts:38]: Oh, I initially wanted the index here but I can just for..of the object. I think simple javascript is well understood by everyone.

File: packages/time-picker/src/TimePicker/TimePicker.tsx [hunk 2]
```diff
@@ -117,17 +117,33 @@ export const TimePicker: React.VFC<TimePickerProps> = ({
...
-  const options = useMemo(
-    () =>
-      getAllTimeOptions({
-        locale,
-        timeZone,
-        date: value,
-        increments: dropdownIncrements,
-      }),
-    [locale, timeZone, dropdownIncrements, value]
+  const [filteredOptions, setFilteredOptions] = React.useState(
```
[the remark this answers is not in the document]
Comment (nadeemaajihil) [reply packages/time-picker/src/TimePicker/TimePicker.tsx:120]: My initial assumption was the component value can be both controlled/uncontrolled so I was treating this filtering separate. Are we sure there shouldn't be an uncontrolled version of this component?

**Comment by HeartSquared** [comment]
Filtering functionality has been revised by designers.
Code for old filtering was also adapted in https://github.com/cultureamp/kaizen-design-system/pull/2977

**Pull request closed without merging**